### PR TITLE
python3Packages.torch: 1.12.1 -> 1.13.0

### PR DESCRIPTION
**Pull request by Jingles52**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
Thanks for your contribution. I went and updated the remaining packages in the ecosystem, because I require them rather quickly to package up the most recent torchaudio source version.

**Comment by ziguana** [comment]
Awesome, thank you! I think you might be missing a sha256 in there somewhere.

**Comment by mweinelt** [comment]
Yeah, *somewhere* being the key word.

File: pkgs/development/python-modules/torchaudio/binary-hashes.nix
Comment (mweinelt) [inline pkgs/development/python-modules/torchaudio/binary-hashes.nix:64]: Here.

**Comment by tpwrules** [comment]
Currently running a full `nixpkgs-review` including CUDA stuff which will take a bit.

It has noticed that `functorch` is now subsumed into the main `pytorch` package and no longer builds: https://github.com/pytorch/functorch/releases/tag/v1.13.0 . I have tested that I can build an environment with just `pytorch` and that `import functorch` succeeds. But I haven't tested any functionality.

I propose simply removing it from nixpkgs and adding a throw to the [aliases file](https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/python-aliases.nix) to say that it is no longer needed. Maybe functorch's maintainer @samuela can weigh in.

**Comment by samuela** [comment]
> It has noticed that functorch is now subsumed into the main pytorch package and no longer builds: https://github.com/pytorch/functorch/releases/tag/v1.13.0 . I have tested that I can build an environment with just pytorch and that import functorch succeeds. But I haven't tested any functionality.
>
> I propose simply removing it from nixpkgs and adding a throw to the [aliases file](https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/python-aliases.nix) to say that it is no longer needed. Maybe functorch's maintainer @samuela can weigh in.

Yup, I think removing and putting an error notice in `python-aliases.nix` is perfect. Also, thank you for running nixpkgs-review @tpwrules !

cc @NixOS/cuda-maintainers

**Comment by tpwrules** [comment]
Result of `nixpkgs-review pr 202769` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

`functorch` needs to be removed as discussed. `pytorch-lightning` is broken due to something unrelated but probably needs an upgrade. `torchinfo` looks like it needs an upgrade too. `pymanopt` builds but doesn't pass tests, the failures look unrelated.

The basic pytorch tests pass, including the CUDA suite. So aside from those nits above this should be good to merge.

<details>
  <summary>6 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.qiskit</li>
    <li>python310Packages.qiskit-machine-learning</li>
    <li>python310Packages.skorch</li>
    <li>python39Packages.qiskit</li>
    <li>python39Packages.qiskit-machine-learning</li>
    <li>python39Packages.skorch</li>
  </ul>
</details>
<details>
  <summary>7 packages failed to build:</summary>
  <ul>
    <li>python310Packages.functorch</li>
    <li>python310Packages.pymanopt</li>
    <li>python310Packages.pytorch-lightning</li>
    <li>python310Packages.torchinfo</li>
    <li>python39Packages.functorch</li>
    <li>python39Packages.pytorch-lightning</li>
    <li>python39Packages.torchinfo</li>
  </ul>
</details>
<details>
  <summary>82 packages built:</summary>
  <ul>
    <li>openai-whisper (python310Packages.openai-whisper)</li>
    <li>python310Packages.arviz</li>
    <li>python310Packages.boxx</li>
    <li>python310Packages.bpycv</li>
    <li>python310Packages.clip</li>
    <li>python310Packages.dalle-mini</li>
    <li>python310Packages.deepwave</li>
    <li>python310Packages.elegy</li>
    <li>python310Packages.ezyrb</li>
    <li>python310Packages.ffcv</li>
    <li>python310Packages.ignite</li>
    <li>python310Packages.monai</li>
    <li>python310Packages.pydmd</li>
    <li>python310Packages.pymc</li>
    <li>python310Packages.pyro-ppl</li>
    <li>python310Packages.pytorch-metric-learning</li>
    <li>python310Packages.pytorch-pfn-extras</li>
    <li>python310Packages.rising</li>
    <li>python310Packages.shap</li>
    <li>python310Packages.skrl</li>
    <li>python310Packages.slicer</li>
    <li>python310Packages.spacy</li>
    <li>python310Packages.spacy-loggers</li>
    <li>python310Packages.spacy-transformers</li>
    <li>python310Packages.stanza</li>
    <li>python310Packages.tensorboardx</li>
    <li>python310Packages.test-tube</li>
    <li>python310Packages.textacy</li>
    <li>python310Packages.torch (python310Packages.torchWithoutCuda)</li>
    <li>python310Packages.torch-tb-profiler</li>
    <li>python310Packages.torchWithCuda</li>
    <li>python310Packages.torchaudio-bin</li>
    <li>python310Packages.torchgpipe</li>
    <li>python310Packages.torchmetrics</li>
    <li>python310Packages.torchvision</li>
    <li>python310Packages.torchvision-bin</li>
    <li>python310Packages.trainer</li>
    <li>python310Packages.treex</li>
    <li>python310Packages.wandb</li>
    <li>python310Packages.zcs</li>
    <li>python39Packages.arviz</li>
    <li>python39Packages.boxx</li>
    <li>python39Packages.bpycv</li>
    <li>python39Packages.clip</li>
    <li>python39Packages.dalle-mini</li>
    <li>python39Packages.deepwave</li>
    <li>python39Packages.elegy</li>
    <li>python39Packages.ezyrb</li>
    <li>python39Packages.ffcv</li>
    <li>python39Packages.ignite</li>
    <li>python39Packages.monai</li>
    <li>python39Packages.openai-whisper</li>
    <li>python39Packages.pydmd</li>
    <li>python39Packages.pymanopt</li>
    <li>python39Packages.pymc</li>
    <li>python39Packages.pyro-ppl</li>
    <li>python39Packages.pytorch-metric-learning</li>
    <li>python39Packages.pytorch-pfn-extras</li>
    <li>python39Packages.rising</li>
    <li>python39Packages.shap</li>
    <li>python39Packages.skrl</li>
    <li>python39Packages.slicer</li>
    <li>python39Packages.spacy</li>
    <li>python39Packages.spacy-loggers</li>
    <li>python39Packages.spacy-transformers</li>
    <li>python39Packages.stanza</li>
    <li>python39Packages.tensorboardx</li>
    <li>python39Packages.test-tube</li>
    <li>python39Packages.textacy</li>
    <li>python39Packages.torch (python39Packages.torchWithoutCuda)</li>
    <li>python39Packages.torch-tb-profiler</li>
    <li>python39Packages.torchWithCuda</li>
    <li>python39Packages.torchaudio-bin</li>
    <li>python39Packages.torchgpipe</li>
    <li>python39Packages.torchmetrics</li>
    <li>python39Packages.torchvision</li>
    <li>python39Packages.torchvision-bin</li>
    <li>python39Packages.trainer</li>
    <li>python39Packages.treex</li>
    <li>python39Packages.wandb</li>
    <li>python39Packages.zcs</li>
    <li>tts</li>
  </ul>
</details>

**Review by junjihashimoto — APPROVED** [review]
LGTM!

**Comment by junjihashimoto** [comment]
By the way, is there a way to fully automate this job?

File: pkgs/top-level/python-aliases.nix
Comment (samuela) [inline pkgs/top-level/python-aliases.nix:85]:
```suggestion
  functorch = throw "functorch is now part of the torch package and has therefore been removed. See https://github.com/pytorch/functorch/releases/tag/v1.13.0 for more info."; # added 2022-12-01
```

Comment (mweinelt) [reply pkgs/top-level/python-aliases.nix:85]: Applied

**Comment by mweinelt** [comment]
> By the way, is there a way to fully automate this job?

For individual packages certainly, for the bunch of them together … difficult.

**Review by junjihashimoto — APPROVED** [review]

**Review by samuela — APPROVED** [review]
LGTM as long as we get a passing nixpkgs-review

**Comment by tpwrules** [comment]
Running one now. Looks like a package got updated so there's a bunch of rebuilding. Will likely post the results tomorrow morning.

**Pull request merged**
